src/meta-client/src/client/ask_leader.rs

```diff
@@ -99,6 +102,28 @@ impl AskLeader {
         Ok(leader)
     }
 
+    pub async fn ask_leader(&self) -> Result<String> {
+        let mut times = 0;
+        while times < self.max_retry {
+            match self.ask_leader_inner().await {
+                Ok(res) => {
+                    return Ok(res);
+                }
+                Err(err) => {
+                    warn!("Failed to ask leader, source: {err}, retry {times} times");
+                    times += 1;
+                    continue;
```

Comment:
Normally retrying is in companion with sleeping. I would suggest using exponential backoff retry. There's an example in procedure framework.

Reply:
It would be better to retry quickly. In k8s, we only configure the address of a load balance of metasrv. There are rare cases where we need to retry, such as DeadlineExceeded, or Unavailable, and it waits until to timeout (10s), so there is no need for additional delay.